### ISSUE TITLE
Remaining timer thread when reconfiguring log4j

### ISSUE THREAD

**Issue by JeromeSoussens** [rfoltyns/log4j2-elasticsearch#15]
Hi,

When log4j2 is reconfigured, a timer thread remains alive. The timer thread comes from the BulkEmitter constructor. 
BulkEmitter#stop() should cancel the timer but with the configuration below it's not the case (stop is never called) : 

```
<Elasticsearch name="elasticsearchAsyncBatch">
            <RollingIndexName indexName="myindex" pattern="yyyy-MM-dd-HH" />
            <AsyncBatchDelivery batchSize="${myBatchSize}" deliveryInterval="1000" >
                <JestHttp serverUris="${myserver}"
                          connTimeout="10000"
                          readTimeout="10000"
                          maxTotalConnection="2"
                          defaultMaxTotalConnectionPerRoute="2"
                          ioThreadCount="2" >
                </JestHttp>
                <AppenderRefFailoverPolicy>
                    <AppenderRef ref="elastic.appender.fallback" level="WARN" />
                </AppenderRefFailoverPolicy>
            </AsyncBatchDelivery>
            <MyCustomLayout />
</Elasticsearch>
```

Code used to reconfigure : 
```
URI uri = URI.create("myConfig.xml");
LoggerContext context = (org.apache.logging.log4j.core.LoggerContext)
                    LogManager.getFactory().getContext("myclassname", null, null, false, uri, "sg-logger");

//... LATER

context.setConfigLocation(uri);
```

Should I change something in the configuration file ?

Thanks in advance,
Jerome

**Comment by rfoltyns** [comment]
I'll clean it up in 1.3.3, cause it's not the only thread that may start in `-core` (see [buffer pool](https://github.com/rfoltyns/log4j2-elasticsearch/blob/master/log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/BufferedItemSourcePool.java)).

Theoretically, only the `ElasticsearchAppender` implements `org.apache.logging.log4j.core.Lifecycle` or `org.apache.logging.log4j.core.Lifecycle2` - I don't remember which one was called during `reconfigure`.
I started to integrate with Log4j2 Lifecycle, but it's not complete yet - just a couple of classes tbh. See [f6e83164](https://github.com/rfoltyns/log4j2-elasticsearch/commit/f6e8316469817ff4f0d8ec29c5516618f4b0576b)
Target picuture is: 
* all necessary `-core` classes with `org.appenders.log4j2.elasticsearch.LifeCycle` (to start decoupling from Log4j2)
* `ElasticsearchAppender` with Log4j2 Lifecycle calling the above (`BatchDelivery`, `BulkEmitter`, clients, pools, etc.)

**Comment by JeromeSoussens** [comment]
Hi,

Thanks for the feedback. 
So I close this issue and I will test 1.3.3  :)

**Comment by foltyr01** [comment]
@JeromeSoussens Thanks for this! I had it in mind for a long time.
I just released a fix for this in 1.3.3.
Enjoy! :)

**Comment by JeromeSoussens** [comment]
Hi @foltyr01 ,
Thanks for that ! 👍 
I'll try it soon.
